extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java

```diff
@@ -177,14 +179,24 @@ public void transform(TransformationContext context) {
                         SpecCompatibility.class,
                         Enablement.class);
 
-        if (metricsCapability.isEmpty()) {
-            builder.addBeanClass("io.smallrye.faulttolerance.metrics.NoopProvider");
-        } else if (metricsCapability.get().metricsSupported(MetricsFactory.MP_METRICS)) {
+        int metricsProviders = 0;
+        if (metricsCapability.isPresent() && metricsCapability.get().metricsSupported(MetricsFactory.MP_METRICS)) {
             builder.addBeanClass("io.smallrye.faulttolerance.metrics.MicroProfileMetricsProvider");
-        } else if (metricsCapability.get().metricsSupported(MetricsFactory.MICROMETER)) {
+            metricsProviders++;
+        } else if (metricsCapability.isPresent() && metricsCapability.get().metricsSupported(MetricsFactory.MICROMETER)) {
             builder.addBeanClass("io.smallrye.faulttolerance.metrics.MicrometerProvider");
+            metricsProviders++;
+        }
+        if (openTelemetrySdk.map(OpenTelemetrySdkBuildItem::isMetricsBuildTimeEnabled).orElse(false)) {
+            builder.addBeanClass("io.smallrye.faulttolerance.metrics.OpenTelemetryProvider");
```

Comment:
This will only be affected by the runtime configuration. We are still going to instrument is the OTel SDK is disabled at runtime...
I suggest you skip the `processFaultToleranceAnnotations()` method in case OpenTelemetry is used and if disabled at runtime.

Reply:
I cannot really do _that_, because I try to do a lot of things at build time and there's no runtime config at that time.

What I _could_ do is read the runtime config and remove the OTEL metrics provider if runtime-disabled. I'm not yet sure what the best way would be -- clearly the `RuntimeValue` is a Quarkus-specific concept and so it cannot be in SmallRye Fault Tolerance. On the other hand, SmallRye Fault Tolerance has configuration for disabling the specific metric providers at runtime, so there might be a way to translate the OTEL config to the SRye FT config, somehow. Not yet sure how.

Reply:
Also, that build item is only produced at runtime, even if it's published with build time data... Not sure if this will work properly and if we need different build items for OTel build and run times.

Reply:
Can the Otel Provider be disabled? 
See: https://github.com/smallrye/smallrye-fault-tolerance/blob/8da7dae36f195608291687793ca9f12fd9567d93/implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/metrics/OpenTelemetryProvider.java#L27

Reply:
Yeah, that's the configuration I mentioned above. There might be a way, I just didn't have the time to explore it yet.